core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -387,6 +362,31 @@ private[deploy] class Master(
   }
 
   override def receiveAndReply(context: RpcCallContext): PartialFunction[Any, Unit] = {
+    case RegisterWorker(
+        id, workerHost, workerPort, workerRef, cores, memory, workerUiPort, publicAddress) => {
+      logInfo("Registering worker %s:%d with %d cores, %s RAM".format(
+        workerHost, workerPort, cores, Utils.megabytesToString(memory)))
+      if (state == RecoveryState.STANDBY) {
+        // ignore, don't send response
```

Comment:
I agree with you, I'm just saying that's a bug in itself, and I'd rather not make that as part of this, because that might affect other parts of the code where the workers retry connections to different masters.